src/main/java/com/epam/test_generator/services/JiraService.java

```diff
@@ -0,0 +1,271 @@
+package com.epam.test_generator.services;
+
+import com.epam.test_generator.dao.impl.JiraProjectDAO;
+import com.epam.test_generator.dao.impl.JiraStoryDAO;
+import com.epam.test_generator.dao.impl.JiraSubStroryDAO;
+import com.epam.test_generator.dao.interfaces.*;
+import com.epam.test_generator.entities.Case;
+import com.epam.test_generator.entities.Project;
+import com.epam.test_generator.entities.Suit;
+import com.epam.test_generator.entities.User;
+import com.epam.test_generator.pojo.JiraProject;
+import com.epam.test_generator.pojo.JiraStory;
+import com.epam.test_generator.pojo.JiraSubTask;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.epam.test_generator.services.utils.UtilsService.checkNotNull;
+
+@Service
+@Transactional
+public class JiraService {
+
+    @Autowired
+    private SuitService suitService;
+
+    @Autowired
+    private CaseService caseService;
+
+    @Autowired
+    private ProjectService projectService;
+
+    @Autowired
+    private JiraSubStroryDAO jiraSubStoryDAO;
+
+    @Autowired
+    private JiraProjectDAO jiraProjectDAO;
+
+    @Autowired
+    private JiraStoryDAO jiraStoryDAO;
+
+    @Autowired
+    private ProjectDAO projectDAO;
+
+    @Autowired
+    private SuitDAO suitDAO;
+
+    @Autowired
+    private CaseDAO caseDAO;
+
+    @Autowired
+    private UserDAO userDAO;
+
+    @Autowired
+    private IssueToDeleteInJiraDAO issueToDeleteInJiraDAO;
+
+    private static final Integer FIRST = 0;
+
+
+    public void createProjectWithAttachments(List<JiraStory> stories) {
+        if (!stories.isEmpty()) {
+
+            String projectKey = stories.get(FIRST).getJiraProjectKey();
+            JiraProject projectByJiraKey = jiraProjectDAO.getProjectByJiraKey(projectKey);
+            createProjectFromJiraProject(projectByJiraKey);
+            addStoriesToExistedProject(stories);
+        }
+    }
+
+    public void addStoriesToExistedProject(List<JiraStory> stories) {
+        stories.forEach(this::createSuitFromJiraStory);
+        String query = String.format(
+                "project=%s AND status in (Open, \"In Progress\", Reopened, Verified) AND type=sub-task",
+                stories.get(FIRST).getJiraProjectKey());
+
+        List<JiraSubTask> jiraSubTasks = jiraSubStoryDAO.getJiraSubtoriesByFilter(query);
+
+        jiraSubTasks.forEach(sub -> {
+            if (suitDAO.findByJiraKey(sub.getJiraParentKey()) != null) {
+                if(caseDAO.findByJiraKey(sub.getJiraKey()) == null)
```

Comment:
Format code, use curly bracers, use either stream api or imperative code (not both):

    jiraSubTasks.stream()
        .filter(subTask -> suitDAO.findByJiraKey(subTask.getJiraParentKey()) != null)
        .filter(subTask -> caseDAO.findByJiraKey(sub.getJiraKey()) == null)
        .forEach(this::createCaseFromJiraSubTask);